src/main/scala/CDS/FileCollection.scala

```diff
@@ -0,0 +1,159 @@
+package CDS
+import scala.io.Source
+import java.net.URI
+
+case class FileCollection(mediaFile:String,inmetaFile:String,metaFile:String,xmlFile:String,dsLocation:URI,tempFile:String){
+  def replace(xmediaFile:Option[String],xinmetaFile:Option[String],xmetaFile:Option[String],xxmlFile:Option[String]):FileCollection = {
+    val newMediaFile = xmediaFile match {
+      case Some(filename)=>filename
+      case None=>mediaFile
+    }
+    val newInmetaFile = xinmetaFile match {
+      case Some(filename)=>filename
+      case None=>inmetaFile
+    }
+    val newMetaFile = xmetaFile match {
+      case Some(filename)=>filename
+      case None=>metaFile
+    }
+    val newXmlFile = xxmlFile match {
+      case Some(filename)=>filename
+      case None=>xmlFile
+    }
+    FileCollection(newMediaFile,newInmetaFile,newMetaFile,newXmlFile,dsLocation,tempFile)
+  }
+
+  def close:Boolean = {
+    try {
+      val f = new java.io.File(tempFile)
+      f.delete()
+      true
+    } catch {
+      case _:Throwable=>
+        false
+    }
+  }
+
+  def getEnvironmentMap(requiredFiles:Seq[String]):Map[String,String] = {
+    if(dsLocation.getScheme != "file") throw new RuntimeException("Non file-based datastores are not yet supported.")
+    requiredFiles.map({
+      case "media"=>Some("cf_media_file"->mediaFile)
+      case "inmeta"=>Some("cf_inmeta_file"->inmetaFile)
+      case "meta"=>Some("cf_meta_file"->metaFile)
+      case "xml"=>Some("cf_xml_file"->xmlFile)
+      case _=>None
+    }).filter({
+      case Some(tuple)=>true
+      case None=>false
+    }).map(
+      _.get
+    ).toMap ++ Map(
+      "cf_datastore_location"->dsLocation.getPath,
+      "cf_temp_file"->tempFile
+    )
+  }
+  
+}
+
+object FileCollection {
+  val tempFileDir=new java.io.File("/tmp/cds_backend")
+
+  implicit class Regex(sc: StringContext) {
```

Comment:
well, i got it from the internets :)

i quite like being able to put the explicit regex text into the case selectors, rather than having to explicitly create objects and then reference them which seems a bit of a drag if you've got a lot of them